clients/githubrepo/branches.go

```diff
@@ -476,23 +505,29 @@ nextRule:
 	return ret, nil
 }
 
+func initializedBoolRef(value bool) *bool {
```

Comment:
```suggestion
func boolPtr(value bool) *bool {
```

Suggested simplification to name. Or at least change to `initializedBoolPtr`, since `*` are usually called "pointers" in the rest of the code ("ref" is only used when talking about git references such as branches, tags, etc).

Reply:
Another idea would be to use generics:

```go
func getPtr[T any](value T) *T {
  // returns a pointer to the given value. Useful for constant values.
  return &value
}
```

Effectively the same, but works for any type. And `getPtr` is better than my previous suggestion since it starts with a verb, making the function's intent clearer.

Reply:
Also liked this idea of using generics here! But in golang we usually avoid using the word `get` in such purposes, so I suggest to keep it as a more meaningful word as `initializePtr` or just `initPtr`.

Reply:
in other places (usually test code) we just do `asPtr`, which would give us
```go
func asPtr[T any](value T) *T {
  // returns a pointer to the given value. Useful for constant values.
  return &value
}
```